emcc.py

```diff
@@ -3767,9 +3767,13 @@ def modularize():
   if not settings.EXPORT_READY_PROMISE:
     return_value = '{}'
 
+  # TODO: Remove when https://bugs.webkit.org/show_bug.cgi?id=223533 is resolved.
+  if async_emit != '' and settings.EXPORT_NAME == 'config':
+    diagnostics.warning('emcc', 'EXPORT_NAME should not be named "config" when targeting Safari')
+
   src = '''
-%(maybe_async)sfunction(%(EXPORT_NAME)s) {
-  %(EXPORT_NAME)s = %(EXPORT_NAME)s || {};
+%(maybe_async)sfunction(config) {
+  var %(EXPORT_NAME)s = config || {};
```

Comment:
Now I'm wondering if we can just put `%(EXPORT_NAME)s = {}` in the parameter list?   Is it possible to want ES6 modules without running on a platform that has default params: https://caniuse.com/mdn-javascript_functions_default_parameters_destructured_parameter_with_default_value_assignment

I guess we can land this either way and possibly follow up.

Reply:
Actually this code is not just for ES6 so lgtm as it.

Reply:
Indeed, this is emitted for all builds that link with `-sMODULARIZE` (which is forcefully enabled when linking with `-sEXPORT_ES6`).